### DISCUSSION TITLE
postgres table ddl

### DISCUSSION THREAD

**Discussion by EurekaJiang** [dbeaver/dbeaver#18003] [Q&A]
There is just no function available to get  the definition of a table in  postgresql. How the DBeaver generate the "DDL"

**Comment by mkgrgis** [comment]
Step by step from different views from https://www.postgresql.org/docs/15/catalogs.html

**Reply by EurekaJiang** [reply]
I mean postgresql provides functions for us to get  the definition of index,view,function, but  just no function available to get the definition of a table. so I need to write the function by myself , when used,   the funtion will be deleted. I wonder if there is a better way to get the  DDL of the table ,so  I  want to know how  it solved in DBeaver

**Reply by mkgrgis** [reply]
> I want to know how it solved in DBeaver

Maybe @LonwoLonwo can give You reference to exact string You need.

**Reply by LonwoLonwo** [reply]
Nope. DBeaver creates DDL partly from the Java side and partly from different SELECTs. You can check all queries in the [Query Manager](https://github.com/dbeaver/dbeaver/wiki/Query-Manager) (do not forget to enable filters)

**Reply by EurekaJiang** [reply]
You’re awesome！
